core/domain/user_services.py

```diff
@@ -1062,6 +1250,81 @@ def set_username(user_id, new_username):
     _save_user_settings(user_settings)
 
 
+def update_user_settings_and_auth_data(user_id, data):
+    """Updates PIN, display alias, subject interests and language preferences
+    for the user with the given user_id.
+
+    Args:
+        user_id: str. The unique ID of the user whose data has to be updated.
+        data: dict. The dictionary containing the new user data, whose keys
+            are the attributes of the user settings and user auth details model.
+    """
+    user_auth_details = get_auth_details_by_user_id(user_id, strict=True)
+    if data['pin'] is None:
+        if user_auth_details.pin is not None:
+            raise utils.ValidationError(
+                'PIN already exists this profile, but provided as empty')
+    else:
+        if not data['pin'] or not isinstance(
+                data['pin'], python_utils.BASESTRING):
+            raise utils.ValidationError(
+                'Expected PIN to be a string, received %s' % data['pin'])
+        user_auth_details.pin = data['pin']
+
+    user_settings = get_user_settings(user_id, strict=True)
+
+    if user_settings.display_alias != data['display_alias']:
+        if (not data['display_alias'] or
+                not isinstance(data['display_alias'], python_utils.BASESTRING)):
+            raise utils.ValidationError(
+                'Expected display alias to be a string, received %s' %
+                data['display_alias']
+            )
+        parent_id = (
+            user_auth_details.parent_user_id if user_auth_details.parent_user_id
+            else user_id
+        )
+        if does_profile_with_display_alias_exist(
+                parent_id, data['display_alias']):
+            raise Exception(
+                'Display alias %s already exists in the account.'
+                % (data['display_alias'])
+            )
+        user_settings.display_alias = data['display_alias']
+
+    if not isinstance(data['subject_interests'], list):
+        raise utils.ValidationError('Expected subject_interests to be a list.')
+    else:
+        for interest in data['subject_interests']:
+            if not isinstance(interest, python_utils.BASESTRING):
+                raise utils.ValidationError(
+                    'Expected each subject interest to be a string.')
+            elif not interest:
+                raise utils.ValidationError(
+                    'Expected each subject interest to be non-empty.')
+            elif not re.match(constants.TAG_REGEX, interest):
+                raise utils.ValidationError(
+                    'Expected each subject interest to consist only of '
+                    'lowercase alphabetic characters and spaces.')
```

Comment:
So based on this we are expecting the data to contain subject interests and preferred language codes etc all to be set ?

Reply:
I think NO for subject interests (will put a condition on top of it -- only deal with it if it's not None).
But seems reasonable to me for languages so YES for that --we set a default value for them during creation. They can only be None or a list of languages defined at the storage (field can only be a enum or None). Also we are not doing any validation checks for them here (similar to how individual functions for these work - `update_preferred_language_codes` etc) so I think code for that part is fine.

Reply:
I guess my question mainly is :Are we assuming that all the information of a profile will be available in the data variable, or just the updated information? From the code it looks like you are expecting all the information?

Reply:
All the information is being passed now. It sounds more appropriate given the decision we have made now of updating all the profiles in go.